src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7351,33 +7351,17 @@ private TypeSyntax ParseTypeCore(ParseTypeMode mode)
             {
                 switch (this.CurrentToken.Kind)
                 {
-                    case SyntaxKind.QuestionToken when canBeNullableType():
+                    case SyntaxKind.QuestionToken:
                         {
-                            var question = EatNullableQualifierIfApplicable(mode);
+                            var question = TryEatNullableQualifierIfApplicable(type, mode);
```

Comment:
ok.  the prior logic was very confusing.  We had 'canBeNullable' here, and then EatNullableQualifierIfApplicable (which also did a bunch *more* canBeNullable checks).  I've merged these all into the renamed TryEatNullableQualifierIfApplicable method now.